torch/_dynamo/source.py

```diff
@@ -260,7 +260,12 @@ def name(self):
         if isinstance(self.index, Source):
             return f"{self.base.name()}[{self.index.name()}]"
         else:
-            return f"{self.base.name()}[{self.index!r}]"
+            if isinstance(self.index, enum.Enum):
+                # Workaround repr(Enum) returns invalid global reference before python 3.11
+                # https://peps.python.org/pep-0663/
+                return f"{self.base.name()}[{self.index!s}]"
```

Comment:
Suggest adding a helper for enum printing.  Can you give an example of what this prints?

Reply:
We prefer ```repr``` rather than ```str``` during codegen, but the behaviors of ```repr``` and ```str``` are opposite compared with other data structures before python 3.11. Enum corrects this behavior at https://peps.python.org/pep-0663/, so we don't need this workaround after 3.11.
```
>>> import enum
>>> class MyEnum(enum.Enum):
...     FOO = 10
...     BAR = 20
... 
>>> repr(MyEnum.FOO)
'<MyEnum.FOO: 10>'
>>> str(MyEnum.FOO)
'MyEnum.FOO'
```
```MyEnum.FOO``` is a valid global reference, so we should use this when codegen.

Reply:
How do we know `MyEnum` is imported in the guard context?